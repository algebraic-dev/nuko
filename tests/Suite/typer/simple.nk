let a: Int -> Int = \f => 2
