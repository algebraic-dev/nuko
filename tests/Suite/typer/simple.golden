✓ That
 └Prod
   ├Program
   | ├typeDecls: Empty List
   | ├letDecls:
   | | └List
   | |   └LetDecl
   | |     ├declName: (ReId a 0:4-0:5)
   | |     ├declArgs: Empty List
   | |     ├declBody:
   | |     | └Lam
   | |     |   ├PId NoExt (ReId f 0:21-0:22)
   | |     |   ├Lit NoExt (LInt 0:26-0:27 2)
   | |     |   └0:20-0:27
   | |     ├declRet: (Type (Prelude.Int -> Prelude.Int))
   | |     └declExt: NoExt
   | ├impDecls: Empty List
   | └programExt: NoExt
   └TypingEnv
     ├_teCurModule: Main
     └_globalTypingEnv:
       └TypeSpace
         ├_tsTypes:
         | └HashMap
         |   ├Prod
         |   | ├Prelude.Int
         |   | └Prod
         |   |   ├Kind *
         |   |   └IsTyDef
         |   └Prod
         |     ├Prelude.String
         |     └Prod
         |       ├Kind *
         |       └IsTyDef
         ├_tsConstructors: HashMap
         ├_tsVars:
         | └HashMap
         |   └Prod
         |     ├Main.a
         |     └Type (Prelude.Int -> Prelude.Int)
         └_tsTypeFields: HashMap
